### ISSUE TITLE
migrate-domain - fails with Git: Unable to read current working directory

### ISSUE THREAD

**Issue by IsNull** [flynn/flynn#2987]
I tried to use my own domain for a simplistic one node test cluster in AWS. The flynn cluster was just setup without any apps installed. Latest stable flynn version `v20160624.1`

I have added two DNS records
`A: cloud.mycompany.com ---> [IP OF FLYNN HOST]`
`CNAME: *.cloud.mycompany.com --> cloud.mycompany.com`

After testing that cloud.mycompany.com really resolves to the flynn host IP, I have started the migration:

`flynn cluster migrate-domain cloud.mycompany.com`

```
Migrate cluster domain from "abcd.flynnhub.com" to "cloud.mycompany.com"? (yes/no): yes
Migrating cluster domain (this can take up to 2m0s)...
Changed cluster domain from "abcd.flynnhub.com" to "cloud.mycompany.com"
error "exit status 128" running "git config --global http.https://git.cloud.mycompany.com.sslCAInfo /Users/isnull/.flynn/ca-certs/flynn-1466842130.pem": "fatal: Unable to read current working directory: No such file or directory\n"
```
The new domain has been properly added to flynn cli and set to default. However, this left the flynn cli in a bad state, since now it has the wrong certifcate/pin for this cluster:

`flynn apps` for example yields now:

`Get https://controller.cloud.mycompany.com/apps: pinned: the peer leaf certificate did not match the provided pin`

PS: I reported this issue just to help you fix the migrate-domain, I already deleted this flynn cluster.

**Comment by titanous** [comment]
Do you happen to have the debug logs from the cluster? The git config step is last, so it shouldn't have had an effect on anything, but the pin error shouldn't be there.

**Comment by IsNull** [comment]
I deleted it already, but let me try again. I report back.

**Comment by IsNull** [comment]
I tried it again, this time the migration did not report any problem:
```
flynn cluster migrate-domain cloud.mycompany.com
Migrate cluster domain from "1dm7.flynnhub.com" to "cloud.mycompany.com"? (yes/no): yes
Migrating cluster domain (this can take up to 2m0s)...
Changed cluster domain from "1dm7.flynnhub.com" to "cloud.mycompany.com"
Updated local CLI configuration
```

```
ElderBook:~ isnull$ flynn cluster
NAME              CONTROLLER URL                          GIT URL                          DOCKER URL
flynn-1467019386  https://controller.cloud.mycompany.com  https://git.cloud.mycompany.com  https://docker.cloud.mycompany.com  (default)
```
So far it looks ok, but the new generated certificate seems not to be included locally:

```
ElderBook:~ isnull$ flynn apps
Get https://controller.cloud.mycompany.com/apps: pinned: the peer leaf certificate did not match the provided pin
```


The flynn-host debug log: https://gist.github.com/anonymous/2b0c17902250c2adc4cc80a5ca3bf9e6

**Comment by jvatic** [comment]
I've reproduced this issue, should have a fix soon

**Comment by jvatic** [comment]
To work around this issue, you'll need to temporarily revert to the old cli config (the new config will work after), then:

Grab the cert and key from the router

```shell
$ flynn -a router env get TLSCERT > cert
$ flynn -a router env get TLSKEY > key
```

Identify the new controller route:

```shell
$ flynn -a controller route
ROUTE                                 SERVICE     ID                                         STICKY  LEADER  PATH
https:controller.cloud.mycompany.com  controller  http/bdcb9363-7b57-45d7-9631-48fb9c3c6390  false   false   /
https:controller.xxxx.flynnhub.com    controller  http/b5e0f13a-d457-4436-a205-68eaf2b01093  false   false   /
```

And update it with the cert and key from the router:

```shell
$ flynn -a controller route update http/bdcb9363-7b57-45d7-9631-48fb9c3c6390 --tls-cert=cert --tls-key=key
updated http/bdcb9363-7b57-45d7-9631-48fb9c3c6390
```

The last two steps will also need to be done for the dashboard and any other apps that should be using this cert.

**Comment by jvatic** [comment]
Here's a script to do the workaround: https://gist.github.com/jvatic/5b0df4daf8490d3076e2d6f7fb75e274
